Dev model

tmsoc opened the pull request:
Here is the initial build for the Model. 

The code still need commenting, but I added a description of all the methods at the top of the file.

The class should have everything we need to get up and running.
Let me know if I should add any more functions.

lnguyen1148 commented:
I understand the functions in our program, just confused a little bit with model class. Are the functions you introduced and the model the same? and the model class is still being developed, isnt it?
I also would like to know precisely which functions that owners can handle, which for administrator?

tmsoc commented:
Linh, the model holds all the functions that access all stored data. Its the middle man between the control and the data being accessed. The control has no direct access to the data. 

I just added comments and fixed a bug in the _sql_to_dict() method. The method would raise an error if the select returned zero records.
